query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryRequestUtil.java

```diff
@@ -90,4 +101,37 @@ public static Expression createStringArrayLiteralValueExpression(List<String> va
                         .setValueType(ValueType.STRING_ARRAY)))
         .build();
   }
+
+  public static boolean isComplexAttributeExpression(Expression expression) {
```

Comment:
This was to ensure that a bad request is not processed further.

Reply:
That should happen during validation though, not nested inside a simple util that's called as part of the query transformation. Also, we'd want to throw if it were a bad query - this would just send it down a different logic path.

Reply:
This is a follow-up task : `Replacing columnIdentifiers with attributeExpressions in preprocessing`

I guess this will remove this confusion ?

Reply:
These are tracked in #100

Reply:
I don't think so - the discussion here is about the type of the attribute rather than whether it comes in a colIdentifier or an attribute expression.

Reply:
Agree here that if we continue to use `isMapAttributeExpression`, it should check:
- if columnIdentifier && isMapField (its attributeId)
- if attributeExpression && isMapFiled (its attributeId)

So, the method will look like as below:
```
 public static boolean isMapAttributeExpression(
      Expression expression, ViewDefinition viewDefinition) {
    return (expression.getValueCase() == COLUMNIDENTIFIER
        && isMapField(expression.getColumnIdentifier().getColumnName(), viewDefinition))
        || (expression.getValueCase() == ATTRIBUTE_EXPRESSION) 
        && isMapField(expression.getAttributeExpression().getAttributeId(), viewDefinition);
  }
```